### PR TITLE
[FIX] website: avoid creating duplicate key for configurator image

### PR DESCRIPTION
**Pull request by SonCrits**
**Description of the issue/feature this PR addresses:**
- Increase user experience in creating website themes
- Eliminate redundant creation of 1 record leading to unnecessary model `ir_model_data` contraints error

**Recreate the Situation**
- Install the website module
- Choose `a business website` in line 1, `fashion designer` in line 2, `get lead` in line 3 on `configurator - step 2`
- Choose random paletee on `configurator - step 3`
- `Build my website` on `configurator - step 4`
- Suppose in this step that we have a bad connection to the api `https://website.api.odoo.com`, the error `duplicate key value violates unique constraint` will appear.

**Reason**
- When a user creates a website, the `configurator_apply` function is called to the api `https://website.api.odoo.com`. If the connection time exceeds 5s, the `configurator_apply` function will be called again. Maximum of 3 times
- Each time `configurator_apply` function runs, will create records of model `ir_model_data` at `set_images` function. This function will create records with a name field like `configurator_1_s_cover_default_image`.
- If this function runs again for the second time, it will also generate a `configurator_1_s_cover_default_image` for the second time. This will result in a `duplicate key value violates unique constraint` error.

**Current behavior before PR:**
- When creating a website theme by industry, if the user's connection to odoo's api has a delay, it will cause an api contraints error of model `ir_model_data`

**Desired behavior after PR is merged:**
- Check if the record of model `ir_model_data` has been created before, if created then skip

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by SonCrits** [comment]
@antonylesuisse please review this PR help me. thank you

**Comment by SonCrits** [comment]
**The `configurator_apply` step callback happens here**

https://github.com/odoo/odoo/blob/f75fda3bba37a5e3e9be3d6f3a11ca49aecb4705/addons/website/static/src/components/configurator/configurator.js#L629-L645

**Comment by SonCrits** [comment]
@rdeodoo please review this pr help me. thank you

**Review by niyasraphy — COMMENTED** [review]
hi @SonCrits ,
seems you haven't signed the CLA yet, thus the runbot test is getting failed.
please add the CLA as per guidelines:  https://github.com/odoo/odoo/blob/15.0/doc/cla/sign-cla.md

thanks

**Comment by SonCrits** [comment]
> 

thank you @niyasraphy

File: addons/website/models/website.py [hunk 1]
```diff
@@ -468,7 +468,14 @@ def configure_page(page_code, snippet_list, pages_views, cta_data):
             page_view_id.save(value=''.join(rendered_snippets), xpath="(//div[hasclass('oe_structure')])[last()]")
 
         def set_images(images):
+            names = self.env['ir.model.data'].search([
+                ('name', '=ilike', 'configurator_%'),
+                ('module', '=', 'website'),
+                ('model', '=', 'ir.attachment')
+            ]).mapped('name')
```
Comment (rdeodoo) [inline addons/website/models/website.py:475]: Didn't check or test but seems like this will always be true for other websites created on the database if the first one went through the configurator?
Need to at least add the `website_id` part in the domain.

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Hello,

Thank you for your contribution.
1. You can remove the `Signed-off-by: Tran Truong Son <truongson290893@gmail.com>` part of your commit message
2. Do you have a detailed step to reproduce the issue so we can directly be able to properly review it? It will also speed up the review process as we don't have to spend time figuring "how".
3. No need to ping multiple people within 1h range, someone will check your PR whenever they got time (Github is on voluntary basis kind of). Note that if it is related to the website app, codeowner should assign the `rd-website` team to your PR. You can also ping us (`@odoo/rd-website`) explicitly in a message if you want to be sure.

**Comment by SonCrits** [comment]
- thank you @rdeodoo  , I have re-added the domain for the `search` . function and add a description of the steps taken to cause the error on the description of the pull request.
 - Please review it for me
 - I have little experience when it comes to contributing to the source code, thanks for your guidance

File: addons/website/models/website.py (hunk 1 above)
Comment (SonCrits) [reply addons/website/models/website.py:475]: @rdeodoo ir_model_data has not related to  website model

**Review by bso-odoo — COMMENTED** [review]
It works great. :+1: Thank you for fixing this.

In the commit message, can you rephrase the title into describing the change ?
E.g. "[FIX] website: avoid creating duplicate key for configurator image".

Here are two small change suggestions:

**Comment by SonCrits** [comment]
> It works great. +1 Thank you for fixing this.
> 
> In the commit message, can you rephrase the title into describing the change ? E.g. "[FIX] website: avoid creating duplicate key for configurator image".
> 
> Here are two small change suggestions:

@bso-odoo , Thank you for reviewing. I have made the necessary revisions. Can you please take a look for me?

**Comment by bso-odoo** [comment]
@SonCrits Oops... it seems the commit now only contains a title. Can you restore a full description below the title ? (feel free to take the text you wrote for the PR description)

File: addons/website/models/website.py [hunk 2]
```diff
@@ -468,7 +468,15 @@ def configure_page(page_code, snippet_list, pages_views, cta_data):
             page_view_id.save(value=''.join(rendered_snippets), xpath="(//div[hasclass('oe_structure')])[last()]")
 
         def set_images(images):
+            names = self.env['ir.model.data'].search([
+                ('name', '=ilike', f'configurator\_{website.id}\_%'),
```
Comment (bso-odoo) [inline addons/website/models/website.py:472]: @SonCrits Oops, my mistake <https://runbot.odoo.com/runbot/batch/1111693/build/48045133>: `\` itself is supposed to be escaped within the f-string.
=> `f'configurator\\_{website.id}\\_%'`
Sorry about this additional iteration.

Comment (SonCrits) [reply addons/website/models/website.py]: thank you bro @bso-odoo

**Comment by bso-odoo** [comment]
@robodoo r+
